### PR TITLE
Mtd 24 method create

### PR DESCRIPTION
**Pull request by ghilyana**
The method create is added on the Controller, Service and Repository layers; Unit tests for these classes are written; Integration test is added.

### CODE REVIEW

File: src/main/java/com/ghilly/service/CityServiceRest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private final CityRepository cityRepository;
+    private final CountryRepository countryRepository;
+
+    public CityServiceRest(CityRepository repository, CountryRepository countryRepository) {
+        this.cityRepository = repository;
+        this.countryRepository = countryRepository;
+    }
+
+    public City create(String cityName, int countryId) {
+        if (!(countryRepository.findById(countryId).isPresent())) throw new IdNotFoundException("The country with the ID " +
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/service/CityServiceRest.java]: exists method isEmpty

Comment (gregorusanov) [reply src/main/java/com/ghilly/service/CityServiceRest.java]: add new line for 'throw new ...'

File: src/main/java/com/ghilly/service/CityServiceRest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                countryId + " is not found.");
+        checkNameIsCorrect(cityName);
+        if (cityRepository.findByName(cityName).isPresent())
```
Comment (gregorusanov) [inline src/main/java/com/ghilly/service/CityServiceRest.java:29]: isEmpty

File: src/test/java/com/ghilly/service/CityServiceRestTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        when(countryRepository.findById(ID)).thenReturn(Optional.of(RUS));
+
+        service.create(NAME, ID);
+
+        assertAll(
+                () -> verify(countryRepository, times(2)).findById(ID),
+                () -> verify(cityRepository).findByName(NAME),
+                () -> verify(cityRepository).save(MOS),
+                () -> verifyNoMoreInteractions(cityRepository),
+                () -> verifyNoMoreInteractions(countryRepository)
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/service/CityServiceRestTest.java]: you can use verifyNoMoreInteractions(cityRepository, cityRepository)

File: src/test/java/com/ghilly/service/CityServiceRestTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import com.ghilly.repository.CountryRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class CityServiceRestTest {
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/service/CityServiceRestTest.java:18]: Where test which checking the condition 'checkNameIsCorrect(cityName);'

File: src/test/java/com/ghilly/integrationTest/CityRestControllerIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+        webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application.properties")
+public class CityRestControllerIntegrationTest {
```
Comment (gregorusanov) [inline src/test/java/com/ghilly/integrationTest/CityRestControllerIntegrationTest.java:31]: Please add one failed test

File: src/main/java/com/ghilly/service/CityServiceRest.java (hunk 1 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CityServiceRest.java]: Done

File: src/main/java/com/ghilly/service/CityServiceRest.java (hunk 2 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CityServiceRest.java:29]: Java advised to use isPresent() :(

File: src/test/java/com/ghilly/service/CityServiceRestTest.java (hunk 3 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/service/CityServiceRestTest.java]: Thanks

File: src/main/java/com/ghilly/service/CityServiceRest.java (hunk 1 above)
Comment (ghilyana) [reply src/main/java/com/ghilly/service/CityServiceRest.java]: Done

File: src/test/java/com/ghilly/service/CityServiceRestTest.java (hunk 3 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/service/CityServiceRestTest.java]: Done

File: src/test/java/com/ghilly/service/CityServiceRestTest.java (hunk 4 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/service/CityServiceRestTest.java:18]: Done

File: src/test/java/com/ghilly/integrationTest/CityRestControllerIntegrationTest.java (hunk 5 above)
Comment (ghilyana) [reply src/test/java/com/ghilly/integrationTest/CityRestControllerIntegrationTest.java:31]: Done

**Review by gregorusanov — APPROVED** [review]

**Pull request merged**